src/jit/lsra.cpp

```diff
@@ -3883,31 +3875,28 @@ void LinearScan::buildRefPositionsForNode(GenTree*                  tree,
         }
 #endif // FEATURE_SIMD
 
-        bool delayRegFree = (hasDelayFreeSrc && useNode->gtLsraInfo.isDelayFree);
         if (useNode->gtLsraInfo.isTgtPref)
         {
             prefSrcInterval = i;
         }
 
-        bool regOptionalAtUse = useNode->IsRegOptional();
-        bool isLastUse        = true;
-        if (isCandidateLocalRef(useNode))
+        regMaskTP fixedAssignment = fixedCandidateMask(type, candidates);
+        if (fixedAssignment != RBM_NONE)
         {
-            isLastUse = ((useNode->gtFlags & GTF_VAR_DEATH) != 0);
+            candidates = fixedAssignment;
         }
-        else
+
+        const bool regOptionalAtUse = useNode->IsRegOptional();
```

Comment:
Not without some other refactoring: its value is derived from `useNode`, which we set to `nullptr` on [line 3897](https://github.com/dotnet/coreclr/pull/10032/files#diff-93fb9832a03a9783e2a415b581fcb45cR3897) before the use of `regOptionalAtUse` on [line 3960](https://github.com/dotnet/coreclr/pull/10032/files#diff-93fb9832a03a9783e2a415b581fcb45cR3960).